src/WebCfgServer.cpp

```diff
@@ -479,13 +479,13 @@ void WebCfgServer::buildHtmlHeader(String &response, String additionalHeader)
     response.concat("<link rel='stylesheet' href='/style.css'>");
     response.concat("<title>Nuki Hub</title></head><body>");
 
-    srand(millis());
+    srand(esp_timer_get_time() / 1000);
 }
 
 void WebCfgServer::waitAndProcess(const bool blocking, const uint32_t duration)
 {
-    unsigned long timeout = millis() + duration;
-    while(millis() < timeout)
+    int64_t timeout = esp_timer_get_time() + (duration * 1000);
```

Comment:
Is it correct that this isn't divided by 1000?

Reply:
Yes because in this case two times in microseconds are compared, where the given duration in milliseconds is multiplied by 1000 to convert to microseconds.